Userland/Applications/Dictionary/Dictionary.cpp

```diff
@@ -0,0 +1,183 @@
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <AK/StringBuilder.h>
+
+#include "Dictionary.h"
+
+namespace Dictionary {
+
+const char* part_of_speech_name(char pos);
+
+const char* part_of_speech_name(char pos)
+{
+    switch (pos) {
+    case 'n':
+        return "noun";
+    case 'v':
+        return "verb";
+    case 's':
+    case 'a':
+        return "adjective";
+    case 'r':
+        return "adverb";
+    default:
+        return "misc";
+    }
+}
+
+Dictionary::Dictionary(const char* filename)
+{
+    auto file_or_error = MappedFile::map(filename);
+    if (file_or_error.is_error()) {
+        m_num_words = 0;
+        m_dictionary_bytes = 0;
+        return;
+    }
+
+    m_dictionary_file = file_or_error.value();
+    m_dictionary_bytes = m_dictionary_file->data();
+
+    const uint32_t* header = (const uint32_t*)m_dictionary_bytes;
+
+    m_num_words = header[0];
+    m_max_word_length = header[1];
+
+    m_index = (decltype(m_index))(
+        (const uint8_t*)m_dictionary_bytes + header[2]);
+
+    m_word_data_index = (decltype(m_word_data_index))(
+        (const uint8_t*)m_dictionary_bytes + header[3]);
+
+    printf("Number of words: %i\n", m_num_words);
```

Comment:
Use `dbgln` or `outln` instead of `printf` here.

Reply:
Thanks! Have now replaced all `printf`s throughout.